### PR TITLE
NativeAot: Enable GC event tracing on standalone GC

### PR DESCRIPTION
**Pull request by AustinWise**
Before this change, no GC events would be recorded when using `dotnet trace` on a NativeAOT program that uses standalone GC.

I copied the implementation [from CoreCLR](https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/gcheaputilities.cpp#L360-L370). The one change I made was to use a Crst instead of a spin lock.

Contributes to #90676

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/gcheaputilities.cpp [hunk 1]
```diff
@@ -47,6 +47,20 @@ extern "C" HRESULT GC_Initialize(
 
 #ifndef DACCESS_COMPILE
 
+CrstStatic g_eventStashLock;
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/gcheaputilities.cpp:50]: Can this be in clrgc.enabled.cpp to make it more pay-for-play?

Comment (AustinWise) [reply src/coreclr/nativeaot/Runtime/gcheaputilities.cpp]: I moved `GCHeapUtilities::RecordEventStateChange` to be defined in both `clrgc.disabled.cpp` and `clrgc.enabled.cpp`. When GC loading is disabled, there is no difference from the status quo.

**Review by jkotas — APPROVED** [review]
Thank you!

@dotnet/gc Could you please sign-off on this change as well?

**Comment by cshung** [comment]
I am experimenting with the change.

**Review by cshung — APPROVED** [review]

**Pull request merged**